src/microbe_stage/organelle_unlocks/IUnlockConstraint.cs

```diff
@@ -0,0 +1,211 @@
+﻿namespace UnlockConstraints
+{
+    using System.Linq;
+
+    /// <summary>
+    ///   A single constraint required for an organelle to be unlocked.
+    ///   These constraints are collected in <see cref="OrganelleUnlockConstraints"/>.
+    /// </summary>
+    public interface IUnlockConstraint
+    {
+        public string Tooltip(GameWorld world, EnergyBalanceInfo energyBalance);
+        public bool Satisfied(GameWorld world, EnergyBalanceInfo energyBalance);
+    }
+
+    /// <summary>
+    ///   The number of microbes engulfed by the player is above the specified amount.
+    /// </summary>
+    public struct EngulfedMicrobesAbove : IUnlockConstraint
```

Comment:
Structs should *only* be used when it really matters for performance. Our default style is that everything is a class unless can shown otherwise that it is needed.